src/components/UserAccess/UserAccessForm/form-utils.ts

```diff
@@ -124,14 +117,7 @@ export const editRB = async (
   const usernames = roleBinding.subjects.map((subject) => subject.name);
   const { role, roleMap } = values;
 
-  // We need to delete the role binding before we create the new one.
-  const existingRBs = await getRBs(roleBinding.metadata.namespace);
-
-  const roleBindingExists = existingRBs.some(
-    (binding) => binding.metadata.name === roleBinding.metadata.name,
-  );
-
-  if (roleBindingExists) {
+  if (!dryRun) {
```

Comment:
I saw in the k8sDeleteResource, we have improved the condition.
Whether we still need to add condition here? what will happen if we do not have the condition?

Reply:
K8sDeleteResource just removed the condition, this condition should be handled by consumer of deleteRB.
There were two things, 
1. if we allowed resource to be deleted during dryrun phase. It tried to refetch the same resource which caused 404 error on API requests
2.DryRun should allow us to catch any errors before hand, so that if user isn't able to create new RB then previous RB still remains and target username doesn't loose previous permissions to the namespace.